Client/Frontend/UserContent/UserScripts/ReadyState.js

```diff
@@ -0,0 +1,77 @@
+// Copyright 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+(function() {
+  // Listen for document ready state
+  document.addEventListener('readystatechange', (event) => {
+    window.webkit.messageHandlers.$<handler>.postMessage({
+      "securitytoken": "$<security_token>",
+      "state": document.readyState
+    });
+  });
+  
+  // Listen for document load state
+  window.addEventListener('load', (event) => {
```

Comment:
small suggestions, but you might remove this event handler after it fires, otherwise a page could trigger the `load` (and similar) events multiple times (using `dispatchEvent`), which will likely cause the backend / swift code to do unexpected things (if, for example, if the backend code expects each page will only `load` once, etc)

Reply:
I need it called more than once for pages that change the document (twitter sometimes calls `window.location.reload(true);`).
If it is called too frequently, the `debounce` will trigger and delay the final call. But if I do need it called only once, I will guard it with a boolean on the Swift side. For now though, I want it this way as we are getting weird behaviours on youtube and other sites that dynamically mess with the history of the page.